backup: refactor some docs for easier understanding (#12338)
pingcap/docs: br/backup-and-restore-overview.md

- Perform history data auditing to meet the requirements of judicial supervision.
- Clone the production environment, which is convenient for troubleshooting, performance tuning, and simulation testing.

## Before you use

> This section describes the prerequisites for using TiDB backup and restore, including the usage tips and compatibility issues.

Author: ```suggestion
This section describes the prerequisites for using TiDB backup and restore, including restrictions, usage tips and compatibility issues.
```

### Restrictions

> PITR:
> 

Reviewer: ```suggestion
```

- PITR does not support restoring the data of user tables or privilege tables from system tables.

> - BR does not support running multiple backup tasks **at a time**.

Reviewer: ```suggestion
- BR does not support running multiple backup tasks on a cluster **at the same time**.
```

> - When a PITR is running, you cannot run a log backup task or use TiCDC to replicate data to a cluster in downstream.

Reviewer: ```suggestion
- When a PITR is running, you cannot run a log backup task or use TiCDC to replicate data to a downstream cluster.
```

### Some tips

Snapshot backup:

- It is recommended that you perform the backup operation during off-peak hours to minimize the impact on applications.

> - It is recommended that you execute multiple backup or restore operations one by one. Running backup operations in parallel leads to low performance. Worse still, lack of collaboration between multiple tasks might result in task failures and affect cluster performance.

Reviewer: ```suggestion
- It is recommended that you execute multiple backup or restore operations one by one. Running multiple backup operations in parallel leads to low performance. Worse still, a lack of collaboration between multiple tasks might result in task failures and affect cluster performance.
```

> 

Reviewer: 这句中文里签名两个任务翻译为 operations，后两个任务翻译为了 tasks，是否可以统一下呢

Author: Done

Snapshot restore:

> - BR uses resources of the target cluster as much as possible. Therefore, it is recommended that you restore data to a new cluster or an offline cluster. Avoid restoring data to a production cluster. Otherwise, your application might be affected.

Reviewer: ```suggestion
- BR uses resources of the target cluster as much as possible. Therefore, it is recommended that you restore data to a new cluster or an offline cluster. Avoid restoring data to a production cluster. Otherwise, your application will be affected inevitably.
```

Backup storage and network configuration:

- It is recommended that you store backup data to a storage system that is compatible with Amazon S3, GCS, or Azure Blob Storage.

> - You need to ensure that BR, TiKV, and the backup storage system have enough network bandwidth, and the storage system can provide sufficient read and write performance (IOPS). Otherwise, they might become a performance bottleneck during backup and restore.

Reviewer: ```suggestion
- You need to ensure that BR, TiKV, and the backup storage system have enough network bandwidth, and that the backup storage system can provide sufficient read and write performance (IOPS). Otherwise, they might become a performance bottleneck during backup and restore.
```
